cuda_phase-retrieval_class/test/cupy_ref.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+template<typename TData>
+class Custom_Cupy_Ref {
+  public :
+    TData * ptr;
```

Comment:
Add a constructor, which takes a py::Object, verify that it is a cupy obj and create a Cupy_Ref.
```c++
auto cp = py::module::import("cupy").attr("array")(v, "dtype"_a="complex128");
Custom_Cupy_Ref ref = Custom_Cupy_Ref(cp);
```